app.js

```diff
@@ -38,6 +54,118 @@ app.post('/vegetables', helpers.validate, (req, res, next) => {
   res.status(201).json(vegetable)
 })
 
+// delete a vegetable
+app.delete('/vegetables/:id', (req, res, next) => {
+  const { vegetables } = data
+  const { id } = req.params
+
+  // if the vegetable exists, delete from the array, return the vegetable
+  if (vegetables.find(veggie => veggie.id === id)) {
+    const vegetable = vegetables.splice(vegetables.findIndex(veggie => veggie.id === id), 1)
+    res.status(200).json(vegetable)
```

Comment:
`.splice()` will return an array, and the expectation here is that this should be an object.